test/api/controllers/stats.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const Promise = require('bluebird');
+
+describe('Stats', () => {
+  before(clearDB);
+
+  afterEach(clearDB);
+
+  describe('GET /v1/stats', () => {
+
+    it('returns the Stats', () => {
+      let promises = [
+        factory.create('trialWithRelated'),
+        factory.create('record'),
+      ];
+      Promise.all(promises).then((result) => {
+        return server.inject('/v1/stats')
+          .then((response) => {
```

Comment:
as you're not using the result from the factories directly, re-arrange this to:

```javascript
return factory.create('trialWithRelated')
  .then(() => factory.create('record'))
  .then(() => server.inject('/v1/stats'))
  .then((response) => {
    // ...
  })
```

The important thing here is to remember to return the promise, otherwise the tests will become flaky because Mocha don't have access to the Promise you're creating, so it can't wait for it to be done. Sometimes it'll be done before Mocha moves on to another test, sometimes it won't.

Also, avoid nested promises when you can. They make the code harder to follow.